### DISCUSSION TITLE
Compiling Webcam & IR function stops with failure?

### DISCUSSION THREAD

**Discussion by RaW79** [benzino77/tasmocompiler#413] [General]
Hi
maybe I missed sth. and try sth. not supported but so far I recognized a very well working lock for not supported selections by Tasmocompiler.

So I like to have ESP32Cam with IR function and made the selection:
![1724231919149-e3d4f1b7-4fa9-4735-93f5-f922fa97298a-grafik](https://github.com/user-attachments/assets/794db519-5ec3-4687-976b-5a75ca65a8d1)

But this ends up in an error because of missing header file IRremoteESP8266.h as follows:

`Compiling .pio/build/tasmota32-webcam/lib4a2/Update/HttpsOTAUpdate.cpp.o
Compiling .pio/build/tasmota32-webcam/lib4a2/Update/Updater.cpp.o
/tmp/Tasmota/tasmota/tasmota_xdrv_driver/xdrv_05_irremote_full.ino:62:10: fatal error: IRremoteESP8266.h: No such file or directory

*************************************************************************
* Looking for IRremoteESP8266.h dependency? Check our library registry!
*
* CLI  > platformio lib search "header:IRremoteESP8266.h"
* Web  > https://registry.platformio.org/search?q=header:IRremoteESP8266.h
*
*************************************************************************

   62 | #include <IRremoteESP8266.h>
      |          ^~~~~~~~~~~~~~~~~~~
compilation terminated.
*** [.pio/build/tasmota32-webcam/src/tasmota.ino.cpp.o] Error 1
========================= [FAILED] Took 77.91 seconds =========================

Environment       Status    Duration
----------------  --------  ------------
tasmota32-webcam  FAILED    00:01:17.910
==================== 1 failed, 0 succeeded in 00:01:17.910 ====================
Finished. Exit code: `1.`

Kindly ask for your check and support to make it happen. Thank you very much in advance.

**Comment by benzino77** [comment]
Hi,
Please follow the links here:
https://github.com/benzino77/tasmocompiler/issues/408

You can also search issues by "webcam" phrase. There is a lot of information about webcam board: https://github.com/benzino77/tasmocompiler/issues?q=webcam

**Reply by RaW79** [reply]
Thanks and understood that ESP32cam is special. Basically no features can be added or deleted by standard so why is this not blocked/locked in the tasmocompiler user interface? Would avoid those questions like mine.

Nevertheless I read also a commend about "frankenstein tasmota". To be honest, I like to control an aircon via IR and need the Webcam to verify from remote that everything is fine by having a look at the aircon display.

Running 2 ESP devices would not be a smart solutions but guess these very very special tasmota versions needs very very special knowdledge and nobody can support a simple tasmocompiler user like, right?

**Reply by benzino77** [reply]
> Thanks and understood that ESP32cam is special. Basically no features can be added or deleted by standard so why is this not blocked/locked in the TasmoCompiler user interface? Would avoid those questions like mine.

Some of the features work ok, some of them not. Some of them are working fine with CAM board today... but will stop working with next Tasmota release. With some tweaks based on configuration from "Custom parameters" it is possible to compile the binary you are thinking about. Will it work? Will it be stable? I have no idea. What I know for sure is that CAM boards are treated special way by Tasmota, and there is probably a reason for that ;) 
Maybe it is a good idea to disable most features when CAM board is chosen and let more advanced users make configuration changes via Custom Parameters? IR and Distance sensors should be disabled for the start....

> very special tasmota versions needs very very special knowledge and nobody can support a simple TasmoCompiler user like, right?

I will encourage you to ask for help on [Tasmota Discord Channel](https://discord.gg/Ks2Kzd4). There are a lot of Tasmota Magicians including Tasmota Dev Team. They are very helpful.

You can also ask for some hints and directions from the author of "frankestein build" -> @andreasbehnke ;) I think all you need are some additional `#defines` in Custom Parameters. 

Another solution is to use something more "beefy" like RPI. I'm pretty sure it will handle all the functionality you need.

**Reply by RaW79** [reply]
> > very special tasmota versions needs very very special knowledge and nobody can support a simple TasmoCompiler user like, right?
> 
> I will encourage you to ask for help on [Tasmota Discord Channel](https://discord.gg/Ks2Kzd4). There are a lot of Tasmota Magicians including Tasmota Dev Team. They are very helpful.

Thank you very much for your constructive reply and really appriciate this. I will try to get in contact and due to I´m not a programmer, just a very happy user of tasmocompiler, i will try to avoid most stupid questions.

> You can also ask for some hints and directions from the author of "frankestein build" -> @andreasbehnke ;) I think all you need are some additional `#defines` in Custom Parameters.

Thanks, I´ll try to get these hints and in parallel I guess I have also understand how to compile tasmota with other tools to make those specialities possible. Thanks to tasmocompiler, this I could avoid so far and had a happy life as non programmer ;)

> Another solution is to use something more "beefy" like RPI. I'm pretty sure it will handle all the functionality you need.

Sure. Due to 24/7 operation hope it´s obvious that the energy consumption of an ESP32 is much more attractive than using a dedicated RPI for this pupose.

**Reply by benzino77** [reply]
I have introduced some changes for esp32cam module to let it be less "special" in TasmoCompiler. It looks like it is now able to compile with features you want to have. 
@RaW79 - please check if "IR feature" is working for you
@andreasbehnke - could you please check if "Distance sensor" is working as you want

To test the changes you need to use `development` version of TasmoCompiler - description how to use it is available [here](https://github.com/benzino77/tasmocompiler/blob/master/README.md#development-version).

**Reply by sfromis** [reply]
Let me remind that the reason for the webcam build being very "thin" on other features is the significant workload for the ESP32 handling the camera, meaning risk of it not being well-working to have the small MCU do other tasks too.

**Reply by RaW79** [reply]
@benzino77 Thank you very much!!! Compiled a development version of webcam including IR and it´s for a quick test running very well. Now I have to solder the IR circuit and test in detail. But I´m very happy and satisfied for your support. It´s clear this is unofficial und therefore might have some problems, but I´ll give it try and my usage is very dedicated to a small application. Without this tryout I´m pretty sure the project we have been not alive any more. Big Thx! As soon as my hardware is fully ready, I´ll give final feedback.
